api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -54,8 +60,21 @@ def on_disconnect(event, context)
 end
 
 def on_default(event, context)
-  sqs = Aws::SQS::Client.new(region: get_region(context))
   message = event["body"]
+  # Return early if this is the user connectivity test
+  if message == 'connectivityTest'
```

Comment:
How risky is it that a user will just hold open a websocket via the connectivity test? Will that cause any issues for us?

Reply:
Maybe we can have the client explicitly disconnect the WebSocket after it has confirmed bidirectional connectivity and also we could explicitly disconnect the connectivity test WebSocket from API Gateway and/or a Lambda if the client doesn't close it within a configured time out?

Reply:
If a user just holds the WebSocket open, there wouldn't really be any effect in usability:
"API Gateway doesn't enforce a quota on concurrent connections. The maximum number of concurrent connections is determined by the rate of new connections per second and maximum connection duration of two hours. For example, with the default quota of 500 new connections per second, if clients connect at the maximum rate over two hours, API Gateway can serve up to 3,600,000 concurrent connections."

Looking at cost:
* WebSocket APIs, the API Gateway free tier includes one million messages (sent or received) and 750,000 connection minutes for up to 12 months.
* $0.25 per million connection minutes
* $1/million messages

One mitigation: The client does explicitly disconnect after a successful connectivity test or an error message.

API Gateway has some default timeouts:
```
Idle Connection Timeout | 10 minutes 
Connection duration for WebSocket API | 2 hours 
```
Would we want to make these tighter?

Reply:
might as well decrease the connection duration since the lambda cannot run for more than 15 minutes anyway. But sounds like it's pretty low risk overall

Reply:
Looks like the API Gateway idle WebSocket timeout is 10 minutes, so we have some basic protection against overconsumption of server resources: https://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html#apigateway-execution-service-websocket-limits-table